code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,336 @@
+			return
+		var/useamount = min(W.amount, (max_plasteel-plasteel_remaining))
+		plasteel_remaining += useamount
+		W.use(useamount)
+		to_chat(user, "<span class='notice'>Inserted [useamount] plateel sheets.")
+		return
+	else
+		to_chat(user, "<span class ='notice'>Invalid material type.</span>")
+	..()
+
+/obj/machinery/computer/camera_advanced/remote_fob/GrantActions(mob/living/user)
+	if(off_action)
+		off_action.target = user
+		off_action.give_action(user)
+		actions += off_action
+
+	if(metal_cade)
+		metal_cade.target = src
+		metal_cade.give_action(user)
+		actions += metal_cade
+
+	if(plast_cade)
+		plast_cade.target = src
+		plast_cade.give_action(user)
+		actions += plast_cade	
+
+	if(toggle_wiring)
+		toggle_wiring.target = src
+		toggle_wiring.give_action(user)
+		actions += toggle_wiring
+
+	if(sentry)
+		sentry.target = src
+		sentry.give_action(user)
+		actions += sentry
+
+	eyeobj.invisibility = 0
+
+/obj/machinery/computer/camera_advanced/remote_fob/remove_eye_control(mob/living/user)
+	.. ()
+	eyeobj.invisibility = INVISIBILITY_ABSTRACT
+	eyeobj.eye_initialized = FALSE
+	return PROCESS_KILL
+
+/obj/machinery/computer/camera_advanced/remote_fob/check_eye(mob/living/user)
+	if(!drone_creation_allowed)
+		to_chat(user, "<span class='notice'>Communication with the drone has been disrupted.</span>")
+		user.unset_interaction()
+	. = ..()
+
+/////////////////////////////// Placement Actions
+
+/datum/action/innate/remote_fob //Parent stuff
+	action_icon = 'icons/Marine/remotefob.dmi'
+	var/mob/living/C //the mob using the action
+	var/mob/camera/aiEye/remote/fobdrone //the drone belonging to the computer
+	var/obj/machinery/computer/camera_advanced/remote_fob/console //the computer itself
+
+/datum/action/innate/remote_fob/Activate()
+	if(!target)
+		return TRUE
+	C = owner
+	fobdrone = C.remote_control
+	console = target
+
+/datum/action/innate/remote_fob/proc/check_spot()
+	var/turf/build_target = get_turf(fobdrone)
+	var/turf/build_area = get_area(build_target)
+
+	if(!is_type_in_typecache(build_area, GLOB.dropship_lzs))
+		to_chat(owner, "<span class='warning'>You can only build within the Landing Zone!</span>")
+		return FALSE
+
+	return TRUE
+
+/datum/action/innate/camera_off/remote_fob
+	name = "Log out"
+
+/datum/action/innate/remote_fob/metal_cade
+	name = "Place Metal Barricade"
+	action_icon_state = "metal_cade"
+	
+
+/datum/action/innate/remote_fob/metal_cade/Activate()
+	if(..() || !check_spot())
+		return
+
+	if(console.metal_remaining < 4)
+		to_chat(owner, "<span class='warning'>Out of material.</span>")
+		return
+
+	var/turf/buildplace = get_turf(fobdrone)
+	var/obj/structure/barricade/cade = /obj/structure/barricade
+	if(is_blocked_turf(buildplace))
+		for(var/obj/thing in buildplace)
+			if(istype(thing, cade) && thing.dir != fobdrone.dir)
+				break
+			else
+				to_chat(owner, "<span class='warning'>No space here for a barricade.</span>")
+				return
```

Comment:
Are you assuming there are more than one cade in this area?
If so, are you not going to output a bunch of messages to chat?

If not, can you not just use `locate()`

Reply:
i initially tried locate here which caused issues, as-is it allows to place 4 cades on the same turf as if you were hand-building them. it only outputs a single message per attempt if the turf wont allow a cade no matter how blocked it is.